src/test/db.ts

```diff
@@ -0,0 +1,303 @@
+                }),
+                headers: nullable(Array),
+                queryFields: nullable(Array),
+                postData: nullable({
+                  mimeType: String,
+                  text: String,
+                }),
+              },
+              checks: nullable(
+                manyOf('assertionText') ||
+                  manyOf('assertionJsonPathValue') ||
+                  manyOf('assertionJsonPath') ||
+                  manyOf('assertionRegex')
+              ),
+            } as any,
+          ]),
+      } as unknown as MultiHttpSettings,
+    } as any,
+  },
+
+  scriptedCheck: {
+    ...baseCheckModel(),
+    settings: {
+      scripted: {
+        script: () => faker.lorem.text(),
+      } as unknown as ScriptedSettings,
+    } as any,
+  },
+
+  browserCheck: {
+    ...baseCheckModel(),
+    settings: {
+      browser: {
+        script: () => faker.lorem.text(),
+      } as unknown as BrowserSettings,
+    } as any,
+  },
+
+  grpcCheck: {
+    ...baseCheckModel(),
+    settings: {
+      grpc: {
+        ipVersion: () => faker.helpers.arrayElement(Object.values(IpVersion)),
+        service: nullable(() => faker.lorem.word()),
+        tls: nullable(() => faker.datatype.boolean()),
+        tlsConfig: nullable(tlsConfig()),
+      } as unknown as GRPCSettings,
+    } as any,
+  },
+
+  probe: {
+    id: primaryKey(() => faker.number.int({ min: 1, max: 999999 })),
+    name: () => faker.string.uuid(),
+    public: () => faker.datatype.boolean(),
+    latitude: () => faker.location.latitude(),
+    longitude: () => faker.location.longitude(),
+    region: () => faker.helpers.arrayElement(['EMEA', 'AMER', 'APAC']),
+    labels: () => [{ name: faker.animal.petName(), value: faker.color.human() } as any],
+    online: () => faker.datatype.boolean(),
+    onlineChange: () => faker.date.past().getTime() / 1000,
+    version: () => faker.system.semver(),
+    deprecated: () => false,
+    modified: () => Math.floor(faker.date.recent().getTime() / 1000),
+    created: () => Math.floor(faker.date.past().getTime() / 1000),
+    capabilities: {
+      disableScriptedChecks: () => faker.datatype.boolean(),
+      disableBrowserChecks: () => faker.datatype.boolean(),
+    } as any,
+  },
+  alert: {
+    id: primaryKey(() => faker.number.int({ min: 1, max: 999999 })),
+    name: () => faker.helpers.arrayElement(Object.values(CheckAlertType)),
+    threshold: () => faker.number.int({ min: 50, max: 500 }),
+    created: () => Math.floor(faker.date.past().getTime() / 1000),
+    modified: () => Math.floor(faker.date.recent().getTime() / 1000),
+  },
+});
+
+export function clean<T>(obj: T): T {
+  const keys = [...Object.getOwnPropertyNames(obj), ...Object.getOwnPropertySymbols(obj)];
+
+  const cleanedObject = keys.reduce((acc, key) => {
+    if (typeof key !== 'symbol') {
+      acc[key as keyof T] = obj[key as keyof T];
+    }
+    return acc;
+  }, {} as T);
+
+  return cleanedObject;
+}
```

Comment:
And can you explain what this is doing and why it is needed?

Reply:
I introduced this `clean` function because the objects created by the mswjs factory contain `Symbol` properties (such as `Symbol(primaryKey): "id"`), which cause mismatches when comparing object equality in tests. This results in failed tests due to these hidden Symbol properties not matching between expected and actual objects. The `clean` function removes these Symbol properties to ensure objects can be compared correctly.

Without this function, tests fail with error messages like the one below:

<img width="261" alt="image" src="https://github.com/user-attachments/assets/af57c660-dd2a-45ce-befe-8ec5229b9208" />

Ideally, I'd prefer to extend the `db.create` function so that we don't have to manually call `clean` or rely on workarounds like this. I'm still exploring the best way to address this issue at the source.